src/mysql_protocol.erl

```diff
@@ -233,26 +270,27 @@ fetch_execute_response(SockModule, Socket, FilterMap, Timeout) ->
     fetch_response(SockModule, Socket, Timeout, binary, FilterMap, []).
 
 %% @doc Changes the user of the connection.
--spec change_user(module(), term(), iodata(), iodata(), binary(),
+-spec change_user(module(), term(), iodata(), iodata(), binary(), binary(),
```

Comment:
Is `AuthPluginName` mandatory for change user or does an empty binary here mean that the default auth method will be used? Comments and/or edoc is welcome (also during development, it's a good habit).

The spec could be easier to read annotated with parameter names. :)

Reply:
Yes, it is required. We need to use the servers' default auth method (like in handshake), but we have no handshake at this point to read it from.

Empty binary appears when the server does not know auth plugins, in which case we resort to mysql_native_password. This is how we handled that before this PR.

Reply:
Ah, I see, the default auth method is stored in the state and passed here, so no change in the exported function for change user. That's good.